### ISSUE TITLE
Compatibility with CTranslate2 / faster-whisper

### ISSUE THREAD

**Issue by entn-at** [huggingface/distil-whisper#3]
Great work!

I was wondering whether the distilled version might still be compatible with CTranslate2 / faster-whisper? I understand the changes to the decoder might require some changes there, not to mention speculative decoding.

Thanks,
Ewald

**Comment by sanchit-gandhi** [comment]
The weights will be released in Transformers format on the Hugging Face Hub tomorrow. It should be pretty straightforward to export them to faster-whisper format following these instructions: https://github.com/guillaumekln/faster-whisper/#model-conversion

I'll add them to the main repo once converted!

**Comment by alexey-mik** [comment]
Unfortunatelly, conversion to CTranslate2 format throws an error
```
ValueError: Some required model attributes are not set:

decoder/layer_2/self_attention/layer_norm/gamma
decoder/layer_2/self_attention/layer_norm/beta
decoder/layer_2/self_attention/linear_0/weight
...
```

**Comment by AnkushMalaker** [comment]
FYI: Related issue on faster-whisper to track full support
https://github.com/guillaumekln/faster-whisper/issues/533

**Comment by chiiyeh** [comment]
Hi! I have done a [PR](https://github.com/OpenNMT/CTranslate2/pull/1529) on Ctranslate2 which will support the conversion for distil-whisper.
Though for the word timing alignment it seems like openai hardcoded the specific cross attention head that are highly correlated with the word timing [here](https://github.com/openai/whisper/blob/v20230306/whisper/__init__.py#L31-L45). Not sure if there is similar one for distil-whisper.

**Comment by chiiyeh** [comment]
@patrickvonplaten  unfortunately not all the cross attentions are highly correlated with word timing. Different cross attention might attend to different things. So what openai did was to find out specifically which of the cross attentions are correlated and only use this subset for the timing alignment.

**Comment by sanchit-gandhi** [comment]
Indeed, OpenAI hardcode these word-level timestamp alignment heads in their repo based on the cross-attention plots.

We haven't found the optimal alignment heads for word-level timestamps for Distil-Whisper, so these word-level timestamps aren't available yet.

Feel free to repeat the analysis from Jong Wook to see what the best configuration is here! We can then update the model's generation config accordingly to store this information. I'll also try and determine the best alignments from the validation sets in Distil-Whisper.

**Comment by shuaijiang** [comment]
> Unfortunately, conversion to CTranslate2 format throws an error
> 
> ```
> ValueError: Some required model attributes are not set:
> 
> decoder/layer_2/self_attention/layer_norm/gamma
> decoder/layer_2/self_attention/layer_norm/beta
> decoder/layer_2/self_attention/linear_0/weight
> ...
> ```

upgrade ctranslate2  to  3.21.0
